### DISCUSSION TITLE
Upload queue in hideable modal

### DISCUSSION THREAD

**Discussion by dro-sh** [rpldy/react-uploady#440] [Q&A] [answered]
I would like to implement modal with queue upload images. I find example with [queue uploading](https://react-uploady.org/docs/guides/UploadQueue/). But I dont understand how to implement it on hideable modal: how to get files that are uploading when modal opens after it has been hidden?

**Comment by yoavniran** [comment, marked as answer]
If you're asking about how to create a component that mounts/dismounts and how to store the data about the uploads then I recommend separating the two concerns. 
The code/component tracking the uploads (ie using the start/progress/finish) and their status should be in a component that doesnt dismount when the modal is hidden. 
Then that data can be passed through props/context (or your choice of state handling).

**Reply by dro-sh** [reply]
Thanks for your answer!
I am using MUI and it Modal component. I thought to store files in redux, but it strongly recomended to not store unserializable objects (like File). So I found Uploady and thought it can hold files in UploadyContext, is not it?

**Reply by yoavniran** [reply]
Why would you need to store the file object in your state? You can store simple values (like the file name, the uploady item ID, etc.)
This way you can store info in your state about the files being uploaded.

You can take a look at this app I built that does something similar (using recoil) - https://github.com/yoavniran/cld-collage-creator/blob/master/src/components/hooks/useUploadyItemMonitor.js

**Reply by dro-sh** [reply]
Ok, thanks! I will try on sandbox
